api/go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/gravitational/teleport/api
 go 1.18
 
 require (
+	github.com/aws/aws-sdk-go v1.44.157
```

Comment:
This came from `api/utils/aws/endpoint.go`, specifically the `ParseDynamoDBServiceFromTarget` function, where I just use 3 constants from the sdk. I can hardcode those as they are unlikely to change, or I can just move that function out of api since it is only referenced by test code which is not necessary.

Reply:
I moved functions that didn't need to be in `api` into the engine itself in `lib` along with the tests for those funcs, removed the `aws-sdk-go` dependency in `api`, and added tests in `api/utils/aws/endpoint_test.go` for `DynamoDBRegionForEndpoint`, because I had apparently forgotten to add a test for that 😬